src/coins.cpp

```diff
@@ -176,8 +176,10 @@ void CCoinsViewCache::SetBestBlock(const uint256 &hashBlockIn) {
     hashBlock = hashBlockIn;
 }
 
-bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlockIn) {
-    for (CCoinsMap::iterator it = mapCoins.begin(); it != mapCoins.end(); it = mapCoins.erase(it)) {
+bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlockIn, bool erase) {
```

Comment:
I'm starting to think we don't need this at all, because the change to `CCoinsViewDB::BatchWrite` is enough.

#15265 had an assert in `CCoinsViewCache::BatchWrite`:
```
assert(erase_from_map); // We do not implement writes to a cache where we don't erase from the child!
```

Reply:
(see my comment here: https://github.com/bitcoin/bitcoin/pull/17487#issuecomment-1405365582)